### PR TITLE
thumbnails for media app

### LINKED ISSUE #14527: Media app file extension icons
In media app add appropriate thumbnails for known file extensions.

before : 

image

after : 

image

image

### PR DESCRIPTION
**Pull request by giannik**
Fixes #14527

### CODE REVIEW

**Review by hishamco — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+﻿// https://github.com/spatie/font-awesome-filetypes
+
+const fa_icons = {
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js:3]: rename this to `faIcons`

File: src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    csv: 'fa-regular fa-file',
+    audio: 'fa-regular fa-file-audio',
+    video: 'fa-regular fa-file-video',
+    archive: 'fa-regular fa-file-zipper',
+    code: 'fa-regular fa-file-code',
+    text: 'fa-regular fa-file-lines',
+    file: 'fa-regular fa-file'
+};
+
+const fa_thumbnails = {
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js:18]: rename to `faThumbnails`

File: src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    gif: fa_icons.image,
+    jpeg: fa_icons.image,
+    jpg: fa_icons.image,
+    png: fa_icons.image,
+
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js:23]: either remove the additional spaces or add a comment in the space.

File: src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+function getExtensionForFilename(filename) {
+    return filename.slice((filename.lastIndexOf('.') - 1 >>> 0) + 2)
+}
+
+
+function getClassNameForFilename(filename) {
+    return getClassNameForExtension(getExtensionForFilename(filename))
+}
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js:68]: add a new line at the end of the file

File: src/OrchardCore.Modules/OrchardCore.Media/Assets.json [hunk 5]
```diff
@@ -12,6 +12,7 @@
       "./node_modules/blueimp-file-upload/js/jquery.fileupload.js",
       "./node_modules/blueimp-file-upload/js/jquery.iframe-transport.js",
       "Assets/js/helpers.js",
+      "Assets/js/fontawesome-icons-thumbnails/**/*.js",
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Media/Assets.json:15]: I don't think you need a new folder for placing this file. You can place it in the MediaApp folder instead.

File: src/OrchardCore.Modules/OrchardCore.Media/Assets/js/app/MediaApp/mediaItemsGridComponent.js [hunk 6]
```diff
@@ -14,7 +14,7 @@ Vue.component('media-items-grid', {
                                 :src="buildMediaUrl(media.url, thumbSize)"
                                 :data-mime="media.mime"
                                 :style="{maxHeight: thumbSize +'px', maxWidth: thumbSize +'px'}" />
-                        <i v-else class="fa-regular fa-file display-1" :data-mime="media.mime"></i>
+                        <i v-else :class="getfontAwesomeClassNameForFileName(media.name, \'fa-6x\')" :data-mime="media.mime"></i>
```
[the remark this answers is not in the document]
Comment (giannik) [reply src/OrchardCore.Modules/OrchardCore.Media/Assets/js/app/MediaApp/mediaItemsGridComponent.js:17]: thats not the point. its for most common file extensions.

File: src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js (hunk 4 above)
Comment (giannik) [reply src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js]: done

File: src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js (hunk 3 above)
Comment (giannik) [reply src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js]: done

File: src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js (hunk 2 above)
Comment (giannik) [reply src/OrchardCore.Modules/OrchardCore.Media/Assets/js/fontawesome-icons-thumbnails/font-awesome-thumbnails.js]: done

**Comment by giannik** [comment]
> @giannik you'll need to also rebuild and commit the changes made to media.js and media.min.js

@MikeAlhayek done

**Review by MikeAlhayek — APPROVED** [review]
LGTM

**Comment by hishamco** [comment]
Is the above code just to change the FA icon for some of the supported extensions :) I don't think it's worth, personally I never mind if others agree of how this could be done

**Comment by giannik** [comment]
for all common file types yes, there is a fallback fa-icon for all other file types

**Review by sebastienros — APPROVED** [review]

**Pull request merged**